Compiler warnings when compiling package with C++ compiler

I am getting a bunch of warnings like the following when compiling the Semigroups package:

 ```In file included from /home/travis/build/gap-packages/gap/src/system.h:27:0,
                 from /home/travis/build/gap-packages/gap/src/compiled.h:17,
                 from src/converter.h:25,
                 from src/semigrp.h:24,
                 from src/semigrp.cc:19:
/home/travis/build/gap-packages/gap/gen/config.h:334:0: warning: "_GNU_SOURCE" redefined
 #define _GNU_SOURCE /**/
<command-line>:0:0: note: this is the location of the previous definition
```
You can see this on travis:

https://travis-ci.org/gap-packages/Semigroups/jobs/243645478

--- a/configure.ac
+++ b/configure.ac
@@ -24,12 +24,6 @@ AC_LANG([C])
 
 
 dnl
-dnl Use _GNU_SOURCE enable extensions like exp10 in glibc
-dnl
-AC_DEFINE([_GNU_SOURCE], [], [Enable GNU extensions in glibc])
-
-
-dnl
 dnl ABI settings
 dnl
 AC_ARG_VAR(ABI, [Set this equal to 32 or 64 to build GAP (and GMP provided you

--- a/src/macfloat.c
+++ b/src/macfloat.c
@@ -11,6 +11,17 @@
 **
 ** macfloats are stored as bags containing a 64 bit value
 */
+
+
+// glibc only declares exp10 in its headers if we define _GNU_SOURCE
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <math.h>
+
+
+
 #include <src/system.h>                 /* system dependent part */
 #include <src/gapstate.h>
 
@@ -42,7 +53,6 @@ extern Double LoadDouble( void);
 extern void SaveDouble( Double d);
 
 #include <stdio.h>
-#include <math.h>
 #include <stdlib.h>
 
 #define SIZE_MACFLOAT   sizeof(Double)

